pkgs/os-specific/darwin/apple-sdk-11.0/cf-setup-hook.sh

```diff
@@ -1,6 +1,14 @@
 forceLinkCoreFoundationFramework() {
-  NIX_CFLAGS_COMPILE="-F@out@/Library/Frameworks${NIX_CFLAGS_COMPILE:+ }${NIX_CFLAGS_COMPILE-}"
-  NIX_LDFLAGS+=" @out@/Library/Frameworks/CoreFoundation.framework/CoreFoundation.tbd"
+  local role_post
+  getHostRole
+
+  local cflags_compile_role_var="NIX_CFLAGS_COMPILE${role_post}"
+  local cflags_compile_role="${!cflags_compile_role_var-}"
+  export NIX_CFLAGS_COMPILE${role_post}="-F@out@/Library/Frameworks${cflags_compile_role:+ $cflags_compile_role}"
+
+  local ldflags_role_var="NIX_LDFLAGS${role_post}"
+  local ldflags_role="${!ldflags_role_var-}"
+  export NIX_LDFLAGS${role_post}+="${ldflags_role:+ }@out@/Library/Frameworks/CoreFoundation.framework/CoreFoundation.tbd"
 }
 
-preConfigureHooks+=(forceLinkCoreFoundationFramework)
```

Comment:
Does this actually need to run on `preConfigureHooks`?

Reply:
It’s been like that since the hook was introduced. That predates my involvement with nixpkgs by quite a bit. It doesn’t seem like it should be necessary.